basedtyping/type_checks.py

```diff
@@ -0,0 +1,18 @@
+from __future__ import annotations
+
+from typing import Generic
+
+from basedtyping.generics import T
+
+
+class assert_type(Generic[T]):
+    """used to assert at type-time that a value is type ``T``
+
+    this is more like a function than a class, but it's defined as a class so that you can explicitly specify the generic"""
+
+    # None return type on __new__ is supported in pyright but not mypy
+    def __new__(cls, value: T) -> None:  # type:ignore[misc]
```

Comment:
Why not init?

Reply:
because it's a hacky fake class thats more like a function, where it doesnt actually return an instance.

it won't be needed once https://github.com/KotlinIsland/basedmypy/issues/121 or https://github.com/KotlinIsland/basedmypy/issues/122 is done